shell/platform/common/text_editing_delta.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
+
+#include <codecvt>
+#include <string>
+
+#include "flutter/shell/platform/common/text_range.h"
+
+namespace flutter {
+
+struct TextEditingDelta {
+ public:
+  TextEditingDelta(std::u16string textBeforeChange,
+                   TextRange range,
+                   std::u16string text);
+  virtual ~TextEditingDelta();
+
+  std::string oldText() const { return utf16ToUtf8(oldText_); }
+  std::string deltaText() const { return utf16ToUtf8(deltaText_); }
+  int deltaStart() const { return deltaStart_; }
+  int deltaEnd() const { return deltaEnd_; }
+
+  TextEditingDelta(std::string textBeforeChange,
+                   TextRange range,
+                   std::string text);
+  TextEditingDelta(std::u16string text);
+  TextEditingDelta(std::string text);
+
+  bool operator==(const TextEditingDelta& rhs) const {
+    return oldText_ == rhs.oldText_ && deltaText_ == rhs.deltaText_ &&
+           deltaStart_ == rhs.deltaStart_ && deltaEnd_ == rhs.deltaEnd_;
+  }
+
+  bool operator!=(const TextEditingDelta& rhs) const { return !(*this == rhs); }
+
+ private:
+  std::u16string oldText_;
+  std::u16string deltaText_;
+  int deltaStart_;
+  int deltaEnd_;
+
+  void setDeltas(std::u16string oldText,
+                 std::u16string newTxt,
```

Comment:
If we do keep this method around, `s/txt/text/` on this, plus switch to `words_separated_by_underscores` form for all of these.